includes/DataObjects/User.php

```diff
@@ -644,7 +644,7 @@ public function isAdmin()
     
 	public function isCheckuser()
 	{
-		return $this->checkuser == 1 || $this->oauthCanCheckUser();
+		return $this->checkuser == 1 || ($this->isOAuthLinked() && $this->oauthCanCheckUser());
```

Comment:
That's what I figured, but the title of the pull request confused me.  Are there any other places we may want to short-circuit OAuth checks in this fashion?